import axios from "axios";
const URL = `${import.meta.env.VITE_API_URL}`;

export const Login = async (payload) => {
    try {
        const response = await axios.post(`${URL}/Nutri`, payload);
        return response.data 
    } catch (error) {
        console.error("Error al a침adir el plan:", error);
    }
};

export const Register = async (payload) => {
    try {
        const response = await axios.post(`${URL}/register`, payload);
        return response.data 
        
    } catch (error) {
        console.error("Error al a침adir registrar el usuario:", error);
    }
};

export const Logout = async () => {
    try {
        const response = await axios.post(`${URL}/logout`, {}, { withCredentials: true });
        localStorage.clear();
        return response.data;
    } catch (error) {
        console.error("Error al cerrar la sesi칩n:", error);
    }
};

export const changePassword = async (currentPassword, newPassword) => {
    try {
        const response = await axios.post(
            `${URL}/change-password`, 
            { currentPassword, newPassword }, 
            { withCredentials: true }
        );
        return response.data;
    } catch (error) {
        console.error("Error al cambiar la contrase침a:", error);
        throw error.response ? error.response.data : { message: "Error desconocido" };
    }
};
